app/components/vendor_outage_alert_component.rb

```diff
@@ -9,7 +9,7 @@ def initialize(vendors:, context: 'default')
   end
 
   def content
-    case outages
+    case outages.sort
```

Comment:
Do we rely on the order for this anywhere? If not, we could just do set comparison, since order it's order independent?

```suggestion
    case outages.to_set # or move the to_set inside the method?
    when Set[:sms, :voice]
    # ...
    when Set[:sms]
    # ...
    when Set[:voice]
```

Reply:
Hm, in an earlier implementation, I had tried to use set, and it wasn't working as expected, though I used the `Set.new` syntax instead of that one. I can give it another try.

Reply:
I just tested this in IRB and I think I might be wrong. My bad!

Reply:
Yeah, even though direct comparison works, it doesn't seem to behave as expected with the `case`:

```rb
outages = [:a, :b].to_set
outages == Set[:b, :a]
# true
case outages
when Set[:b, :a]
  'unordered'
when Set[:a, :b]
  'same order'
else
  'neither'
end
# "neither"
```

Reply:
The biggest bummer:
```
Set[1, 2] === Set[1, 2]
# false
```

but it reminded me that `===` for sets is a membership test so if we really wanted to do sets we could do this (but I don't recommend)

```
case outages.to_set
when Set[Set[:sms, voice]]
  # ...
when Set[Set[:sms]]
  # ...
when Set[Set[:voice]]
  # ..
end
```

Reply:
The order doesn't matter except to make sure we produce the same message regardless if vendors is given as `[:voice, :sms]` or `[:sms, :voice]`. I also recently changed this in e0c9339 to make it a bit more localized to the case block, to try to further reinforce this point that it only matters in this set of conditions (some more detail in extended commit message).

Reply:
I've also been contemplating whether this component could handle _many_ alert messages, which I expect could be implemented very differently and avoid this sort.

Something like:

```rb
alerts = []
while outages.present? do
  if outages.include?(:sms) && outages.include?(:voice)
    alerts << t(context, scope: 'vendor_outage.alerts.phone', default: :default)
    outages -= [:sms, :voice]
  elsif outages.include?(:sms)
    alerts << t(context, scope: 'vendor_outage.alerts.sms', default: :default)
    outages -= [:sms]
  elsif outages.include?(:voice)
    alerts << t(context, scope: 'vendor_outage.alerts.voice', default: :default)
    outages -= [:voice]
  elsif outages.include?(:lexisnexis_trueid)
    # ...
  else
    outages = []
  end
end
```